src/battle_ai_main.c

```diff
@@ -2417,11 +2458,19 @@ static s16 AI_TryToFaint(u8 battlerAtk, u8 battlerDef, u16 move, s16 score)
     
     if (CanAttackerFaintTarget(battlerAtk, battlerDef, AI_THINKING_STRUCT->movesetIndex, 0) && gBattleMoves[move].effect != EFFECT_EXPLOSION)
     {
-        // this move can faint the target
-        if (GetWhoStrikesFirst(battlerAtk, battlerDef, TRUE) == 0 || GetMovePriority(battlerAtk, move) > 0)
-            score += 4; // we go first or we're using priority move
+        // Heavily force kill move.
+        // Prioritize killing moves with highest priority.
+        if (GetMovePriority(battlerAtk, move) > 0)
+            score += 10 + 2 * GetMovePriority(battlerAtk, move);
         else
-            score += 2;
+            score += 10;
+
+        // Not very good, albeit still useable.
+        if (moveEffect == EFFECT_RECHARGE)
+            score -= 5;
+
+        // this move can faint the target
+        score += 4;
```

Comment:
this seems redundant. isn't the `score += 10` under the `else` statement basically doing the same thing?

Reply:
True, it's a leftover I forgot to remove. Corrected.